evm_loader/program/src/types/transaction.rs

```diff
@@ -266,13 +266,123 @@ impl rlp::Decodable for AccessListTx {
     }
 }
 
-// TODO: Will be added as a part of EIP-1559
-// struct DynamicFeeTx {}
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DynamicFeeTx {
+    pub nonce: u64,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub max_priority_fee_per_gas: U256,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub max_fee_per_gas: U256,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub gas_limit: U256,
+    pub target: Option<Address>,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub value: U256,
+    #[serde(with = "serde_bytes")]
+    pub call_data: Vec<u8>,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub r: U256,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub s: U256,
+    #[serde(with = "ethnum::serde::bytes::le")]
+    pub chain_id: U256,
+    pub recovery_id: u8,
+    pub access_list: Vec<(Address, Vec<StorageKey>)>,
+}
+
+impl rlp::Decodable for DynamicFeeTx {
+    fn decode(rlp: &rlp::Rlp) -> Result<Self, rlp::DecoderError> {
+        let rlp_len = {
+            let info = rlp.payload_info()?;
+            info.header_len + info.value_len
+        };
+
+        if rlp.as_raw().len() != rlp_len {
+            return Err(rlp::DecoderError::RlpInconsistentLengthAndData);
+        }
+
+        let chain_id: U256 = u256(&rlp.at(0)?)?;
+        let nonce: u64 = rlp.val_at(1)?;
+
+        let max_priority_fee_per_gas: U256 = u256(&rlp.at(2)?)?;
+        let max_fee_per_gas: U256 = u256(&rlp.at(3)?)?;
+        if max_fee_per_gas < max_priority_fee_per_gas {
+            return Err(rlp::DecoderError::Custom(
+                "max_fee_per_gas < max_priority_fee_per_gas",
+            ));
+        }
+
+        let gas_limit: U256 = u256(&rlp.at(4)?)?;
+        let target: Option<Address> = {
+            let target = rlp.at(5)?;
+            if target.is_empty() {
+                if target.is_data() {
+                    None
+                } else {
+                    return Err(rlp::DecoderError::RlpExpectedToBeData);
+                }
+            } else {
+                Some(target.as_val()?)
+            }
+        };
+
+        let value: U256 = u256(&rlp.at(6)?)?;
+        let call_data = rlp.val_at(7)?;
+
+        let rlp_access_list = rlp.at(8)?;
+        let mut access_list = vec![];
+
+        for entry in &rlp_access_list {
+            // Check if entry is a list
+            if entry.is_list() {
+                // Parse address from first element
+                let address: Address = entry.at(0)?.as_val()?;
+
+                // Get storage keys from second element
+                let mut storage_keys: Vec<StorageKey> = vec![];
+
+                for key in &entry.at(1)? {
+                    storage_keys.push(key.as_val()?);
+                }
```

Comment:
Maybe I'm reading this wrong, but since `entry` doesn't look like it's changing, isn't this an infinite loop?

Reply:
`entry` is being "changed" by the outer loop, right? Here we just iterate over the inner list of addresses. Maybe I'm missing something?:)

**Off-topic, some context around access list state in Neon:**

You see, AccessList transactions are supported by our EVM for quite a long time.

The problem, they are **NOT** supported by the proxy (operator never creates such transactions and always falls back to creation of legacy ones). 
Moreover, they are **NOT** very well supported even by Ethereum's infra. For example, Metamask is yet to add creation of such transactions: https://github.com/MetaMask/metamask-extension/issues/11863

Summing things up: AccessList transactions mechanism is not battle tested at all :) What I did is basically copied this piece of rlp parsing from access list transactions into dynamic gas transactions. It's quite possible there are  some bugs that should be discovered during testing.

P.S. Also, last piece of my long rant: access list is an Ethereum mechanism designed to save some gas given transaction specifies addresses/memory cells it's going to touch. For Neon, I can't see how it plays out simply because in solana operator is being charged per-signature (and not per memory/address access).